sdk/cosmos/azure_data_cosmos/src/models/container_properties.rs

```diff
@@ -0,0 +1,402 @@
+use std::time::Duration;
+
+use azure_core::Model;
+use serde::{Deserialize, Deserializer};
+
+use crate::models::SystemProperties;
+
+#[cfg(doc)]
+use crate::clients::ContainerClientMethods;
+
+fn deserialize_ttl<'de, D>(deserializer: D) -> Result<Option<Duration>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    Ok(Option::<u64>::deserialize(deserializer)?.map(Duration::from_secs))
+}
+
+/// Properties of a Cosmos DB container.
+///
+/// Returned by [`ContainerClient::read()`](crate::clients::ContainerClient::read()).
+#[non_exhaustive]
+#[derive(Model, Clone, Debug, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct ContainerProperties {
```

Comment:
You could consider adding the [`vectorEmbeddingPolicy`](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/4f515bb56f6687b61e780bc77f306b24b09fde7b/Microsoft.Azure.Cosmos/src/Resource/Settings/VectorEmbeddingPolicy.cs#L21) stuff here.  It's preview only in .NET right now, but already in some of the other SDKs.

Reply:
I'm going to hold on that for now, I don't yet have an account set up where I can verify that this data is properly being deserialized from the server's response, so I'll look into that and then add support for it.

Reply:
No problem - though to set up the account, it's just a toggle on the features blade: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/vector-search#enroll-in-the-vector-search-preview-feature

Reply:
(But that can defnitely come later)

Reply:
Ah yeah, I actually have messed with that before in a different account. I can probably add it in quick.

Reply:
It was indeed pretty straightforward. I've added it now:

```
vector_embedding_policy: Some(
        VectorEmbeddingPolicy {
            embeddings: [
                VectorEmbedding {
                    path: "/vector1",
                    data_type: Float32,
                    dimensions: 10,
                    distance_function: Euclidean,
                },
                VectorEmbedding {
                    path: "/vector2",
                    data_type: Uint8,
                    dimensions: 10,
                    distance_function: Cosine,
                },
                VectorEmbedding {
                    path: "/vector3",
                    data_type: Int8,
                    dimensions: 4096,
                    distance_function: DotProduct,
                },
            ],
        },
    ),
```